src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php

```diff
@@ -0,0 +1,272 @@
+        $openBracesCount = 0;
+        $isNamespaceWithBraces = false;
+        $isWithinNamespace = false;
+
+        for ($index = 0, $count = $tokens->count(); $index < $count; ++$index) {
+            $token = $tokens[$index];
+
+            $tokenContent = $token->getContent();
+
+            if ('{' === $tokenContent) {
+                ++$openBracesCount;
+
+                continue;
+            }
+
+            if ('}' === $tokenContent) {
+                --$openBracesCount;
+
+                if (true === $isWithinNamespace
+                    && true === $isNamespaceWithBraces
+                    && 0 === $openBracesCount
+                ) {
+                    $isWithinNamespace = false;
+                }
+
+                continue;
+            }
+
+            if ($token->isGivenKind(T_NAMESPACE)) {
+                $namespaceDeclarationEndIndex = $tokens->getNextTokenOfKind($index, array(
+                    ';',
+                    '{',
+                ));
+
+                if ($tokens[$namespaceDeclarationEndIndex]->getContent() === '{') {
+                    $isNamespaceWithBraces = true;
+                }
+
+                $isWithinNamespace = true;
+            }
+
+            if (!$isWithinNamespace) {
+                continue;
+            }
+
+            // test if we are at a function call
+            if (!$token->isGivenKind(T_STRING)) {
+                continue;
+            }
+
+            $next = $tokens->getNextMeaningfulToken($index);
+            if (!$tokens[$next]->equals('(')) {
+                continue;
+            }
+
+            $functionNamePrefix = $tokens->getPrevMeaningfulToken($index);
+            if ($tokens[$functionNamePrefix]->isGivenKind(array(T_DOUBLE_COLON, T_NEW, T_OBJECT_OPERATOR, T_FUNCTION))) {
+                continue;
+            }
+
+            if ($tokens[$functionNamePrefix]->isGivenKind(T_NS_SEPARATOR)) {
+                // skip if the call is to a constructor or to a function in a namespace other than the default
+                $prev = $tokens->getPrevMeaningfulToken($functionNamePrefix);
+                if ($tokens[$prev]->isGivenKind(array(T_STRING, T_NEW))) {
+                    continue;
+                }
+            }
+
+            $lowerFunctionName = \strtolower($tokenContent);
+
+            if (!\in_array($lowerFunctionName, $internalFunctionNames, true) || \in_array($lowerFunctionName, $this->exclude, true)) {
+                continue;
+            }
+
+            // do not bother if previous token is already namespace separator
+            if ($tokens[$index - 1]->isGivenKind(T_NS_SEPARATOR)) {
+                continue;
+            }
+
+            $indexes[] = $index;
+        }
+
+        $indexes = \array_reverse($indexes);
+
+        $namespaceSeparator = new Token(array(
+            T_NS_SEPARATOR,
+            '\\',
+        ));
+
+        foreach ($indexes as $index) {
+            $tokens->insertAt($index, $namespaceSeparator);
```

Comment:
Nice approach to not iterate backwards.
I would put `array_reverse` directly before `foreach` look to emphasise that we do insert in reverse order.
 
But, you must not insert same element at different indexes. This is token object. If one will decide in another fixer to change one of them, he will change all of them instead.